mcnp_input_reader.py

```diff
@@ -223,29 +308,196 @@ def cells_with_mat(mat_num, cell_list):
+            surf_clean.append(line)
+            if line == '':
+                surf_clean.pop(i)
+
+    # Assumes that a surface description that goes over one line
+    # will have a whitespace at the end of every continuing line.
+    bloc = []
+    comments_list = []
+    surf_new = []
+    for i, row in enumerate(surf_clean):
+        if row[0] == ' ':
+            new_row = surf_clean[i-1] + surf_clean[i]
+            surf_new.append(new_row)
+            try:
+                surf_new.pop(i-1)
+            except IndexError:
+                continue
+        else:
+            surf_new.append(row)
+
+    # finds the comments and pops them out and into another list
+    for i, row in enumerate(surf_new):
+        data, sep, comment = row.partition(' $')
+        bloc.append(data)
+        if comment == '':
+            comments_list.append('--')
+        else:
+            comments_list.append(comment)
+
+    # joins the row separated list
+    # splits it on spaces, now there is a list
+    bloc_str = ' \n '.join(bloc)
+    bloc_sep = bloc_str.split(' ')
+
+    # gets rid of empty entries ''
+    surf = []
+    [surf.append(c) for c in bloc_sep if c != '']
+
+    # back into one string, but line separated with \n from earlier
+    # so it can be split based on \n
+    joined = ' '.join(surf)
+    split_l = joined.splitlines()
+
+    # now splits on whitespace and then appends to new list so every
+    # interesting bit of data is an entry in a list of lists
+    new_list = []
+    for i in range(len(split_l)):
+        new_list.append(split_l[i].split(' '))
+
+    # list with no ''
+    sdata = []
+    for i, row in enumerate(new_list):
+        row = list(filter(None, new_list[i]))
+        sdata.append(row)
+
+    # init lists for dict to go to dataframe
+    s_num = []
+    s_type = []
+    s_transform = []
+    s_params = []
+
+    # adds data to columns, special case for transform
+    for i, c in enumerate(sdata):
+        s_num.append(c[0])
+        if float_check(c[1]) is True:
+            s_transform.append(c[1])
+            s_type.append(c[2])
+            s_params.append(c[3:])
+        else:
+            s_transform.append('--')
+            s_type.append(c[1])
+            s_params.append(c[2:])
 
+    dict_df = {'Num': s_num, 'Transform': s_transform, 'Type': s_type,
+               'Parameters': s_params, 'Comments': comments_list}
+    df = pd.DataFrame(dict_df)
+
+    return df
 
-def read_mcnp_input(fpath):
-    """ """
 
+def read_mcnp_input(fpath, input):
+    """ reads and returns input object"""
     ifile = ut.get_lines(fpath)
     cell_bloc, surf_bloc, data_bloc = split_blocs(ifile)
-    cell_list = process_cell_block(cell_bloc)
+    cell_ls = process_cell_block(cell_bloc)
+    surf_df = surface_reader(surf_bloc)
+    mat_nums = get_material_numbers(data_bloc)
+    tally_nums = get_tally_numbers(data_bloc)
+    input.filename = fpath
+    input.cell_list = len(cell_ls)

```

Comment:
think these should be the actual not the lengths

Reply:
i think this is what you print and should be just the lengths for now